File: AlterAmbient/FormEliminarLista.cs
using System;
using System.Data;
using System.Windows.Forms;
using IAHRIS.BBDD;
using IAHRIS.Rellenar;
using Microsoft.VisualBasic.CompilerServices;

namespace IAHRIS
{
    public partial class FormEliminarLista
    {
        public FormEliminarLista()
        {
            InitializeComponent();
            _lstboxListas.Name = "lstboxListas";
            _btnAltDiaria.Name = "btnAltDiaria";
            _btnBorrarNatDiaria.Name = "btnBorrarNatDiaria";
            _btnBorrarNatMensual.Name = "btnBorrarNatMensual";
            _cmboxAlt.Name = "cmboxAlt";
            _btnAltMensual.Name = "btnAltMensual";
        }

        public FormEliminarLista(OleDbDataBase MDB)
        {

            // This call is required by the Windows Form Designer.
            InitializeComponent();

            // Add any initialization after the InitializeComponent() call.
            _cMDB = MDB;
            _Rellenar = new Rellenar.RellenarForm(_cMDB);

            // -------------------------------------
            // ---- Traducir formulario ------------
            // -------------------------------------
            Form argform = this;
            _traductor = new MultiLangXML.MultiIdiomasXML(ref argform);
            _traductor.traducirFormPorConf(Application.StartupPath, @"\conf.xml");
            _lstboxListas.Name = "lstboxListas";
            _btnAltDiaria.Name = "btnAltDiaria";
            _btnBorrarNatDiaria.Name = "btnBorrarNatDiaria";
            _btnBorrarNatMensual.Name = "btnBorrarNatMensual";
            _cmboxAlt.Name = "cmboxAlt";
            _btnAltMensual.Name = "btnAltMensual";

            
           
        }

        private Rellenar.RellenarForm _Rellenar;
        private BBDD.OleDbDataBase _cMDB;
        private int _idpunto;
        private int _codAlt;
        private MultiLangXML.MultiIdiomasXML _traductor;

        private void FormEliminarLista_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.F1)
            {
                // Help.ShowHelp(Me, "AlterHidrol.chm", HelpNavigator.KeywordIndex, "Eliminar serie")
            }
        }

        private void FormEliminarLista_Load(object sender, EventArgs e)
        {
            var argcombo = cmbProyectos;
            _Rellenar.RellenarProyectos(ref argcombo);
            
        }

        private void lboxListas_SelectedIndexChanged(object sender, EventArgs e)
        {
            DataSet ds;
            DataRow draux;
            btnBorrarNatDiaria.Enabled = false;
            btnBorrarNatMensual.Enabled = false;
            btnAltDiaria.Enabled = false;
            btnAltMensual.Enabled = false;
            //ds = _cMDB.RellenarDataSet("Valor", "SELECT id_punto FROM [Punto] WHERE Id_Punto=" + ((ComboItem)lstboxListas.SelectedItem).Id );
            //draux = ds.Tables[0].Rows[0];
            _idpunto = ((ComboItem)lstboxListas.SelectedItem).Id;//Conversions.ToInteger(draux[0]);
            // Sacar nombre y id
            if (lstboxListas.SelectedIndex != -1)
            {
                ds = _cMDB.RellenarDataSet("Valor", "SELECT * FROM [Lista] WHERE ID_punto=" + _idpunto.ToString());
                // dr = ds.Tables(0).Rows(0)
                foreach (DataRow dr in ds.Tables[0].Rows)
                {
                    // Borrar lista natural diaria
                    if (Conversions.ToBoolean(Operators.AndObject(dr["Tipo_Lista"], dr["Tipo_Fechas"])))
                    {
                        btnBorrarNatDiaria.Enabled = true;
                    }
                    else if (Conversions.ToBoolean(Operators.AndObject(dr["Tipo_Lista"], !(bool)dr["Tipo_Fechas"])))
                    {
                        btnBorrarNatMensual.Enabled = true;
                    }
                }
            }

            var argcombo = cmboxAlt;
            _Rellenar.RellenarAlteraciones(ref argcombo, _idpunto);
            cmboxAlt = argcombo;
        }

        private void cmboxAlt_SelectedIndexChanged(object sender, EventArgs e)
        {
            DataSet ds;
            DataRow draux;
            btnAltDiaria.Enabled = false;
            btnAltMensual.Enabled = false;
            //ds = _cMDB.RellenarDataSet("Valor", "SELECT id_alteracion FROM [Alteracion] WHERE COD_Alteracion='" + cmboxAlt.SelectedItem.ToString() + "'");
            //draux = ds.Tables[0].Rows[0];
            _codAlt = ((ComboItem)cmboxAlt.SelectedItem).Id;// Conversions.ToInteger(draux[0]);
            ds = _cMDB.RellenarDataSet("Valor", "SELECT * FROM [Lista] WHERE ID_punto=" + _idpunto.ToString() + " AND id_alteracion=" + _codAlt.ToString());
            // dr = ds.Tables(0).Rows(0)
            foreach (DataRow dr in ds.Tables[0].Rows)
            {
                // Borrar lista natural diaria
                if (Conversions.ToBoolean(dr["Tipo_Fechas"]))
                {
                    btnAltDiaria.Enabled = true;
                }
                else if (Conversions.ToBoolean(!(bool)dr["Tipo_Fechas"]))
                {
                    btnAltMensual.Enabled = true;
                }
            }
        }

        private void btnBorrarNatDiaria_Click(object sender, EventArgs e)
        {
            _cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_punto = " + _idpunto + " AND Tipo_Lista=TRUE AND Tipo_fechas=TRUE AND id_alteracion=0");
            lboxListas_SelectedIndexChanged(lstboxListas, null);
        }

        private void btnBorrarNatMensual_Click(object sender, EventArgs e)
        {
            _cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_punto = " + _idpunto + " AND Tipo_Lista=TRUE AND Tipo_fechas=FALSE AND id_alteracion=0");
            lboxListas_SelectedIndexChanged(lstboxListas, null);
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            // Dim idLista As Integer

            // Dim ds As DataSet
            // Dim dr As DataRow

            // ' Sacar nombre y id
            // If (Me.lstboxListas.SelectedIndex <> -1) Then
            // ds = Me._cMDB.RellenarDataSet("Valor", "SELECT id_lista FROM Lista WHERE nombre='" & Me.lstboxListas.SelectedItem.ToString() & "'")

            // dr = ds.Tables(0).Rows(0)

            // idLista = Integer.Parse(dr("id_lista"))

            // Me._cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_lista = " & idLista)

            // Me._Rellenar.RellenarListas(Me.lstboxListas, -1)
            // End If
            _cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_punto = " + _idpunto + " AND Tipo_Lista=FALSE AND Tipo_fechas=TRUE AND id_alteracion=" + _codAlt.ToString());
            lboxListas_SelectedIndexChanged(lstboxListas, null);
        }

        private void btnAltMensual_Click(object sender, EventArgs e)
        {
            _cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_punto = " + _idpunto + " AND Tipo_Lista=FALSE AND Tipo_fechas=FALSE AND id_alteracion=" + _codAlt.ToString());
            lboxListas_SelectedIndexChanged(lstboxListas, null);
        }

        private void cmbProyectos_SelectedIndexChanged(object sender, EventArgs e)
        {
            var arglistbox = lstboxListas;
            _Rellenar.RellenarPuntos(ref arglistbox, ((ComboItem)cmbProyectos.SelectedItem).Id);
            lstboxListas = arglistbox;
            btnBorrarNatDiaria.Enabled = false;
            btnBorrarNatMensual.Enabled = false;
            btnAltDiaria.Enabled = false;
            btnAltMensual.Enabled = false;
            Left = (int)(0.5d * (Screen.PrimaryScreen.Bounds.Width - Width));
            Top = (int)(0.5d * (Screen.PrimaryScreen.Bounds.Height - Height));
        }
    }
}

File: AlterAmbient/FormEliminarLista.vb
Imports IAHRIS.Rellenar.RellenarForm

Public Class FormEliminarLista

    Dim _Rellenar As Rellenar.RellenarForm
    Dim _cMDB As BBDD.OleDbDataBase

    Dim _idpunto As Integer
    Dim _codAlt As Integer

    Private _traductor As MultiLangXML.MultiIdiomasXML

    Public Sub New(ByVal MDB As BBDD.OleDbDataBase)

        ' This call is required by the Windows Form Designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        Me._cMDB = MDB

        Me._Rellenar = New Rellenar.RellenarForm(Me._cMDB)

        ' -------------------------------------
        ' ---- Traducir formulario ------------
        ' -------------------------------------
        Me._traductor = New MultiLangXML.MultiIdiomasXML(Me)
        Me._traductor.traducirFormPorConf(Application.StartupPath, "\conf.xml")
    End Sub

    Private Sub FormEliminarLista_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.F1 Then
            'Help.ShowHelp(Me, "AlterHidrol.chm", HelpNavigator.KeywordIndex, "Eliminar serie")
        End If
    End Sub

    Private Sub FormEliminarLista_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

        Me._Rellenar.RellenarPuntos(Me.lstboxListas)
        Me.btnBorrarNatDiaria.Enabled = False
        Me.btnBorrarNatMensual.Enabled = False
        Me.btnAltDiaria.Enabled = False
        Me.btnAltMensual.Enabled = False

        Me.Left = 0.5 * (System.Windows.Forms.Screen.PrimaryScreen.Bounds.Width - Me.Width)
        Me.Top = 0.5 * (System.Windows.Forms.Screen.PrimaryScreen.Bounds.Height - Me.Height)
    End Sub

    Private Sub lboxListas_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles lstboxListas.SelectedIndexChanged

        Dim ds As DataSet
        Dim draux As DataRow

        Me.btnBorrarNatDiaria.Enabled = False
        Me.btnBorrarNatMensual.Enabled = False
        Me.btnAltDiaria.Enabled = False
        Me.btnAltMensual.Enabled = False

        ds = Me._cMDB.RellenarDataSet("Valor", "SELECT id_punto FROM [Punto] WHERE Clave_Punto='" & Me.lstboxListas.SelectedItem.ToString() & "'")

        draux = ds.Tables(0).Rows(0)
        Me._idpunto = draux(0)
        ' Sacar nombre y id
        If (Me.lstboxListas.SelectedIndex <> -1) Then
            ds = Me._cMDB.RellenarDataSet("Valor", "SELECT * FROM [Lista] WHERE ID_punto=" & Me._idpunto.ToString())
            'dr = ds.Tables(0).Rows(0)
            For Each dr As DataRow In ds.Tables(0).Rows
                ' Borrar lista natural diaria
                If (dr("Tipo_Lista") And dr("Tipo_Fechas")) Then

                    Me.btnBorrarNatDiaria.Enabled = True

                ElseIf (dr("Tipo_Lista") And (Not dr("Tipo_Fechas"))) Then

                    Me.btnBorrarNatMensual.Enabled = True

                End If
            Next
        End If

        Me._Rellenar.RellenarAlteraciones(Me.cmboxAlt, Me._idpunto)

    End Sub

    Private Sub cmboxAlt_SelectedIndexChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles cmboxAlt.SelectedIndexChanged
        Dim ds As DataSet

        Dim draux As DataRow

        Me.btnAltDiaria.Enabled = False
        Me.btnAltMensual.Enabled = False

        ds = Me._cMDB.RellenarDataSet("Valor", "SELECT id_alteracion FROM [Alteracion] WHERE COD_Alteracion='" & Me.cmboxAlt.SelectedItem.ToString() & "'")

        draux = ds.Tables(0).Rows(0)

        Me._codAlt = draux(0)

        ds = Me._cMDB.RellenarDataSet("Valor", "SELECT * FROM [Lista] WHERE ID_punto=" & Me._idpunto.ToString() & " AND id_alteracion=" & Me._codAlt.ToString())
        'dr = ds.Tables(0).Rows(0)
        For Each dr As DataRow In ds.Tables(0).Rows
            ' Borrar lista natural diaria
            If (dr("Tipo_Fechas")) Then
                Me.btnAltDiaria.Enabled = True
            ElseIf (Not dr("Tipo_Fechas")) Then
                Me.btnAltMensual.Enabled = True
            End If
        Next
    End Sub

    Private Sub btnBorrarNatDiaria_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnBorrarNatDiaria.Click
        Me._cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_punto = " & Me._idpunto & " AND Tipo_Lista=TRUE AND Tipo_fechas=TRUE AND id_alteracion=0")
        lboxListas_SelectedIndexChanged(Me.lstboxListas, Nothing)
    End Sub

    Private Sub btnBorrarNatMensual_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnBorrarNatMensual.Click
        Me._cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_punto = " & Me._idpunto & " AND Tipo_Lista=TRUE AND Tipo_fechas=FALSE AND id_alteracion=0")
        lboxListas_SelectedIndexChanged(Me.lstboxListas, Nothing)
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAltDiaria.Click
        'Dim idLista As Integer

        'Dim ds As DataSet
        'Dim dr As DataRow

        '' Sacar nombre y id
        'If (Me.lstboxListas.SelectedIndex <> -1) Then
        '    ds = Me._cMDB.RellenarDataSet("Valor", "SELECT id_lista FROM Lista WHERE nombre='" & Me.lstboxListas.SelectedItem.ToString() & "'")

        '    dr = ds.Tables(0).Rows(0)

        '    idLista = Integer.Parse(dr("id_lista"))

        '    Me._cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_lista = " & idLista)

        '    Me._Rellenar.RellenarListas(Me.lstboxListas, -1)
        'End If
        Me._cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_punto = " & Me._idpunto & " AND Tipo_Lista=FALSE AND Tipo_fechas=TRUE AND id_alteracion=" & Me._codAlt.ToString())
        lboxListas_SelectedIndexChanged(Me.lstboxListas, Nothing)
    End Sub

    Private Sub btnAltMensual_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnAltMensual.Click
        Me._cMDB.EjecutarSQL("DELETE * FROM Lista WHERE id_punto = " & Me._idpunto & " AND Tipo_Lista=FALSE AND Tipo_fechas=FALSE AND id_alteracion=" & Me._codAlt.ToString())
        lboxListas_SelectedIndexChanged(Me.lstboxListas, Nothing)
    End Sub
End Class